src/extension/sldService/src/main/java/org/geoserver/sldservice/utils/classifier/RulesBuilder.java

```diff
@@ -470,4 +476,29 @@ private Expression normalizeProperty(
     private boolean isDuplicatedClass(List<Rule> rules, Filter f) {
         return rules.stream().anyMatch(r -> r.getFilter().equals(f));
     }
+
+    /**
+     * Compares current min and previous min avoiding the production of overlapping Rules
+     *
+     * @param currentIdx
+     * @param groups
+     * @param att
+     * @return
+     */
+    private Filter getNotOverlappingFilter(
+            int currentIdx, RangedClassifier groups, Expression att, List<Rule> list) {
+        Filter f;
+        if (currentIdx > 0) {
+            Rule prevRule = list.get(list.size() - 1);
+            Object currMin = groups.getMin(currentIdx);
+            Object prevMin = groups.getMin(currentIdx - 1);
+            if (!prevMin.equals(currMin) || !(prevRule.getFilter() instanceof PropertyIsEqualTo))
```

Comment:
Isn't this redundant? If the prev min is the same as the current min, the previous filter must include the value... or not? Asking cause I'm not sure, but if it does, you could reduce the list of arguments of this method.

Reply:
Indeed it is, removed the check as well as the param.